lib/src/main/kotlin/io/wavebeans/lib/SampleArray.kt

```diff
@@ -0,0 +1,41 @@
+package io.wavebeans.lib
+
+import io.wavebeans.lib.stream.window.Window
+
+typealias SampleVector = DoubleArray
+
+val EmptySampleVector: SampleVector = SampleVector(0)
+
+@Suppress("NOTHING_TO_INLINE")
+fun sampleVectorOf(list: List<Sample>): SampleVector {
+    val i = list.iterator()
+    return SampleVector(list.size) { i.next() }
+}
+
+@Suppress("NOTHING_TO_INLINE")
+fun sampleVectorOf(vararg sample: Sample): SampleVector = SampleVector(sample.size) { sample[it] }
+
+@Suppress("NOTHING_TO_INLINE")
+fun sampleVectorOf(window: Window<Sample>): SampleVector = sampleVectorOf(window.elements)
```

Comment:
add function to extract the sample vector out of the window. Perhaps use sample vector for internal window representation.